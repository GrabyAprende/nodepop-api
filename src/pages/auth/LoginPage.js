import { useState } from "react";
import storage from "../../utils/storage";
import { useDispatch } from "react-redux";
import { loginThunk, setToken } from "../../store/actions/sessionActions";
import { useLogin } from "../../hooks/useLogin";

function LoginPage() {
  const dispatch = useDispatch();
  const rememberMeData = storage.get("nodePopCredentials");
  const login = useLogin();

  const [email, setEmail] = useState(
    rememberMeData ? rememberMeData.email : ""
  );
  const [password, setPassword] = useState(
    rememberMeData ? rememberMeData.password : ""
  );
  const [rememberMe, setRememberMe] = useState(false);

  const handleSubmit = async (event) => {
    event.preventDefault();
    dispatch(loginThunk({ email, password, rememberMe }));

    const token = await login({
      email,
      password,
      rememberMe,
    });

    dispatch(setToken(token));
  };

  const handleEmailChange = (event) => {
    setEmail(event.target.value);
  };
  const handlePasswordChange = (event) => {
    setPassword(event.target.value);
  };
  const handleRememberChange = (event) => {
    setRememberMe(event.target.checked);
  };

  const disabled = !(email && password);

  return (
    <section className="container">
      <article>
        <h1>Login to Nodepop</h1>
        <form onSubmit={handleSubmit}>
          <input
            type="text"
            name="email"
            onChange={handleEmailChange}
            value={email}
          />
          <input
            type="password"
            name="password"
            onChange={handlePasswordChange}
            value={password}
          />
          <fieldset>
            <label htmlFor="remember">
              <input
                type="checkbox"
                role="switch"
                id="remember"
                name="remember"
                onChange={handleRememberChange}
              />
              Remember me
            </label>
          </fieldset>
          <button type="submit" disabled={disabled}>
            Log in
          </button>
        </form>
      </article>
    </section>
  );
}

export default LoginPage;
