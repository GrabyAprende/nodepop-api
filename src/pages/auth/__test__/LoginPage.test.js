// LoginPage.test.js
import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import LoginPage from "../LoginPage";

const mockStore = configureStore([]);
const store = mockStore({});

describe("LoginPage", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <LoginPage />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
