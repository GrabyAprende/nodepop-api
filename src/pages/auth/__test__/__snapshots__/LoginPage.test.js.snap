// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders correctly 1`] = `
<section
  className="container"
>
  <article>
    <h1>
      Login to Nodepop
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <fieldset>
        <label
          htmlFor="remember"
        >
          <input
            id="remember"
            name="remember"
            onChange={[Function]}
            role="switch"
            type="checkbox"
          />
          Remember me
        </label>
      </fieldset>
      <button
        disabled={true}
        type="submit"
      >
        Log in
      </button>
    </form>
  </article>
</section>
`;
